### PR TITLE
Fixing outstanding linting issues (SCP-3267)

### PR DESCRIPTION
**Pull request by devonbush**
SCP-3267 This fixes all outstanding eslint errors and warnings for our javascript.  

Screen Shot 2021-04-14 at 1 52 56 PM

The config change of disabling `no-lonely-if` is because I often find nested ifs are clearer than else-ifs when the conditions are unrelated  e.g.

```
if (foo == 5) {
  // do foo stuff
} else {
  if (bar) {
    // do bar stuff 
  }
}
```

but happy to change that if you'd prefer the other way.

### CODE REVIEW

File: app/javascript/components/search/results/Study.js [hunk 1]
```diff
@@ -46,30 +46,44 @@ export function shortenDescription(textDescription, term) {
...
+      <span className = 'openingText'>{displayedBeginningText} </span>
+      <span className="detail">... </span>
+      <span className = 'studyDescription' dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}
+    </>
   }
   const displayedStudyDescription = { __html: styledText.slice(0, descriptionCharacterLimit) }
   if (textDescription.length>descriptionCharacterLimit) {
-    return <><span className = 'studyDescription' dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}</>
+    return <>
+      <span className = 'studyDescription' dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}
```
Comment (eweitz) [inline app/javascript/components/search/results/Study.js:85]:
```suggestion
      <span className="studyDescription" dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}
```

File: app/javascript/lib/metrics-api.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ export function logClickLink(target) {
     id: target.id
   }
   // Check if target is a tab that's not a part of a menu
-  const parentTabList = $(target).closest("[data-analytics-name][role='tablist']")
+  const parentTabList = $(target).closest('[data-analytics-name][role=\'tablist\']')
```
Comment (eweitz) [inline app/javascript/lib/metrics-api.js:126]: ESLint substantially degrades readability in this case, so let's disable it here.

```suggestion
  // eslint-disable-next-line
  const parentTabList = $(target).closest("[data-analytics-name][role='tablist']")
```

File: app/javascript/lib/scp-api.js [hunk 3]
```diff
@@ -244,22 +244,25 @@ export async function fetchClusterOptions(studyAccession, mock=false) {
...
  * @param {String} subsample Subsampling threshold, e.g. 100000
  * @param {String} consensus Statistic to use for consensus, e.g. "mean"
  * @param {Boolean} isAnnotatedScatter If showing "Annotated scatter" plot.
  *                  Only applies for numeric (not group) annotations.
  * @param {Boolean} mock If using mock data.  Helps development, tests.
  *
  * Example:
- * https://localhost:3000/single_cell/api/v1/studies/SCP56/clusters/Coordinates_Major_cell_types.txt?annotation_name=CLUSTER&annotation_type=group&annotation_scope=study
+ * https://localhost:3000/single_cell/api/v1/studies/SCP56/clusters/
+     Coordinates_Major_cell_types.txt?annotation_name=CLUSTER&annotation_type=group&annotation_scope=study
```
Comment (eweitz) [inline app/javascript/lib/scp-api.js:265]: ESLint also substantially degrades readability here, so let's pass over this case.

```suggestion
 * // eslint-disable-next-line
 * https://localhost:3000/single_cell/api/v1/studies/SCP56/clusters/Coordinates_Major_cell_types.txt?annotation_name=CLUSTER&annotation_type=group&annotation_scope=study
```

**Review by eweitz — CHANGES_REQUESTED** [review]
Nice cleanup.  Let's tidy a few more nits (and nits-of-nits, since it's a a linting PR).

File: app/javascript/lib/scp-api.js (hunk 3 above)
Comment (eweitz) [reply app/javascript/lib/scp-api.js:265]: Thoughts on this, @devonbush?

**Review by bistline — APPROVED** [review]

**Review by eweitz — APPROVED** [review]
Addressing [this straggler](https://github.com/broadinstitute/single_cell_portal_core/pull/983#discussion_r613523398) would be nice (but I won't hold things up over it!).

**Pull request merged**